app/javascript/packs/mock-device-profiling.tsx

```diff
@@ -0,0 +1,124 @@
+ * import.meta.url doesn't have live URL params so we need to scrape the DOM.
+ */
+function loadSessionId(): string | undefined {
+  let sessionId;
+  Array.from(document.scripts).every((scriptTag) => {
+    if (scriptTag.src.includes('session_id')) {
+      sessionId = new URL(scriptTag.src).searchParams.get('session_id');
+      return false;
+    }
+    return true;
+  });
+  return sessionId;
+}
+
+function submitMockFraudResult({ result, sessionId }: { result: string; sessionId?: string }) {
+  window.fetch('/test/device_profiling', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      session_id: sessionId,
+      result,
+    }),
+  });
+}
+
+const CHAOS_OPTIONS: { apply: () => void; undo: () => void }[] = [
+  {
+    apply: () => {
+      document.body.style.transform = 'scale(-1, 1)';
+    },
+    undo: () => {
+      document.body.style.transform = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.transform = 'rotate(5deg)';
+    },
+    undo: () => {
+      document.body.style.transform = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.filter = 'invert(100%)';
+    },
+    undo: () => {
+      document.body.style.filter = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.fontFamily = 'Comic Sans MS';
+    },
+    undo: () => {
+      document.body.style.fontFamily = '';
+    },
+  },
+];
+
+let undoLast: () => void | undefined;
+
+function mockFraudResultSelected(event: ChangeEvent) {
+  if (!(event.target instanceof HTMLSelectElement)) {
+    return;
+  }
+
+  const { value } = event.target;
+
+  if (value === 'chaotic') {
+    const randomChaosOption = CHAOS_OPTIONS[Math.floor(Math.random() * CHAOS_OPTIONS.length)];
+    randomChaosOption.apply();
+    undoLast = randomChaosOption.undo;
+  } else {
+    undoLast?.();
+    submitMockFraudResult({ result: value, sessionId: loadSessionId() });
+  }
+}
+
+function MockDeviceProfilingOptions() {
+  const instanceId = useInstanceId();
+  const inputId = `select-input-${instanceId}`;
+
+  return (
+    <>
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+      <label className="usa-label" htmlFor={inputId}>
+        <strong className="text-accent-warm-dark">For sandbox testing only:</strong> Mock device
+        profiling behavior
+      </label>
+      <select
+        className="border-05 border-accent-warm-dark"
+        onChange={mockFraudResultSelected}
+        name="mock_profiling_result"
+        id={inputId}
+      >
+        <option value="no_result">No Result</option>
+        <option value="pass">Pass</option>
+        <option value="reject">Reject</option>
+        <option value="review">Review</option>
+        <option value="chaotic">Do something chaotic</option>
+      </select>
+    </>
+  );
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  const ssnInput = document.getElementsByName('doc_auth[ssn]')[0];
```

Comment:
Might be nice to output some debugging info to the console here:

```suggestion
  console.log(`Mock device profiling started for session ID ${loadSessionId()}`);

  const ssnInput = document.getElementsByName('doc_auth[ssn]')[0];
```

Reply:
We have some ESLints that discourage console logging, and we have the UI element for the dropdown, would this be like to help us detect if the DOM elements we look for move and the UI fails to load?

Reply:
I think I'm going to skip this:
- we have code in our feature specs that fails if it notices console.log output (which we usually only get in the case of errors) so adding an exception for this on every SSN page would be noisy
- I added steps to the feature specs for the SSN step that select debug options from here and verify the result in redis, so if the JS breaks, our test suite should tell us

so I think we have guards that cover the things this helps with already